### PR TITLE
Adjust srml, Add reference obs documentation

### LINKED ISSUE #436: Fix SRML sites
Some SRML sites lack observations, in some cases this is due to lack of data. The following sites should be removed from the initialization csv:
- Dillon MT: https://dashboard.solarforecastarbiter.org/observations/?site_id=9717b3ca-7e49-11e9-b2d0-0a580a8003e9
- Boise ID : https://dashboard.solarforecastarbiter.org/observations/?site_id=9bd092c6-7e49-11e9-bb52-0a580a8003e9
- Green River WY: https://dashboard.solarforecastarbiter.org/observations/?site_id=9a818e66-7e49-11e9-a9cd-0a580a8003e9
- Moab UT: https://dashboard.solarforecastarbiter.org/observations/?site_id=9a69d8a6-7e49-11e9-b080-0a580a8003e9

Sites with data that failed to initialize observations:
- Silver Lake OR: https://dashboard.solarforecastarbiter.org/observations/?site_id=99a07366-7e49-11e9-a5f5-0a580a8003e9
- Portland OR PV: https://dashboard.solarforecastarbiter.org/observations/?site_id=96662e54-7e49-11e9-ba57-0a580a8003e9
- UO Solar Awning Eugene OR: https://dashboard.solarforecastarbiter.org/observations/?site_id=93f5edb4-7e49-11e9-bfe2-0a580a8003e9

Updating these probably just means deleting them from production, and reinitializing them from a partial csv.

In the process of looking through missing data I've found some overlooked PV data available from srml at the following sites: 
http://solardat.uoregon.edu/PortlandPV.html
http://solardat.uoregon.edu/UOSolarAwning.html (might be difficult to parse, multiple angles)
http://solardat.uoregon.edu/BendPV.html

### PR DESCRIPTION
**Pull request by lboeman**
- [x] Closes #436  .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [x] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

This PR does a few things:

- Removes SRML sites that haven't reported data for a while. 
- When initializing SRML sites, use a data file from 30 days ago to ensure there's actually a file. Due to the delay, some of the current SRML sites in prod don't have observations.
- Add power observations for Ashland OR, Kalapuya HS, Bend OR, and Portland OR PV. I've added a PV suffix to the end of each of their names. I've used the json format that was introduced in #438, for the PV sites.
- Adds a top level "reference observations" page to the documents as suggested in #429. I've tried to update the explanation from that PR. I think it still needs work.

### CODE REVIEW

File: docs/source/reference-observations.rst [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+   the site, (e.g. when a PV plant requires `modeling_parameters`) this function
+   may be used to update the site metadata before it is posted to the API.
+    * site: dict
+
+
+Available Network Handlers
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* NOAA (The National Oceanic and Atmospheric Administration)
+   * SURFRAD: Surface Radiation Budget Network
```
Comment (wholmgren) [inline docs/source/reference-observations.rst:103]: For consistency with DOE and NREL, don't group together the NOAA networks. I don't have a good understanding of the history of any of them, but I'm guessing they might as well be at different institutions anyways, so a flat list seems appropriate from that perspective too.

File: docs/source/whatsnew/1.0.0rc1.rst [hunk 2]
```diff
@@ -30,13 +30,17 @@ Enhancements
 * Reports can now be rendered into PDF files via pdflatex (:issue:`319`) (:pull:`439`)
 * Report generation via the CLI can now run through mock API endpoints to test
   report serialization (:issue:`405`) (:pull:`441`)
+* Adjusted University of Oregon Solar Radiation Measurment Laboratory (UO SRML)
+  to include available pv power data. (:pull:`442`)
```
Comment (wholmgren) [inline docs/source/whatsnew/1.0.0rc1.rst:34]:
```suggestion
  to include available PV power data. (:pull:`442`)
```

File: docs/source/whatsnew/1.0.0rc1.rst [hunk 3]
```diff
@@ -30,13 +30,17 @@ Enhancements
...
 
 Bug fixes
 ~~~~~~~~~
 * Fix incorrect ordering of months and weekdays in metrics plots.
   (:issue:`428`) (:pull:`430`)
 * Ensure data is sorted from reference data sources before slicing and
   posting to the API (:pull:`435`)
+* Remove UO SMRL sites from reference dataset that have not reported recent
+  data. (:issue:`436`)
```
Comment (wholmgren) [inline docs/source/whatsnew/1.0.0rc1.rst:43]:
```suggestion
  data. (:issue:`436`) (:pull:`442`)
```

File: solarforecastarbiter/io/reference_observations/srml.py [hunk 4]
```diff
@@ -36,17 +46,38 @@
 logger = logging.getLogger('reference_data')
 
 
+def adjust_site_parameters(site):
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_observations/srml.py:49]: looks like nearly a copy of the `pvdaq` function. should we refactor into common?

**Comment by wholmgren** [comment]
There were a handful of warnings related to the new files when building the docs. Check the readthedocs build log or build locally (``make -C docs html``)

File: solarforecastarbiter/io/reference_observations/tests/test_srml.py [hunk 5]
```diff
@@ -29,8 +92,89 @@ def test_fetch(mocker, single_site, start, end, exp):
...
+    logger = mocker.patch(
+        'solarforecastarbiter.io.reference_observations.srml.iotools.'
+        'logger.warning')
+    get_func = mocker.patch(
+        'solarforecastarbiter.io.reference_observations.srml.iotools.'
+        'read_srml_month_from_solardat')
+    get_func.side_effect = exception()
+    data = srml.request_data(test_site, 1, 1)
+    assert logger.call_count == 5
+    assert data is None
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_observations/tests/test_srml.py:180]: Are these tests actually run? They don't start with `test_` so I would assume not. Also codecov says no.

File: solarforecastarbiter/io/reference_observations/srml.py (hunk 4 above)
Comment (alorenzo175) [reply solarforecastarbiter/io/reference_observations/srml.py:49]: Yes.  Minimal changes would be to move the meat to common with a sitefile kwarg then just call that from pvdaq/srml with the right sitefile

**Comment by lboeman** [comment]
I factored out the modelling parameter adjustment code into common.py, but I can't figure out what's causing the FileNotFound error in the tests. I've copied the behavior used for the tests in `io.fetch` that have their own test data. @alorenzo175 any ideas on what's wrong here?

**Comment by alorenzo175** [comment]
I'm not seeing the test file at all. Did it git ignored and not committed?

File: solarforecastarbiter/io/reference_observations/pvdaq.py [hunk 6]
```diff
@@ -41,17 +41,7 @@ def adjust_site_parameters(site):
...
-    with open(DEFAULT_SITEFILE) as fp:
-        sites_metadata = json.load(fp)['sites']
-    site_api_id = int(site['extra_parameters']['network_api_id'])
-    for site_metadata in sites_metadata:
-        site_extra_params = json.loads(site_metadata['extra_parameters'])
-        if site_extra_params['network_api_id'] == site_api_id:
-            site_out = site.copy()
-            site_out['modeling_parameters'] = site_metadata[
-                'modeling_parameters']
-            site_out['extra_parameters'].update(site_extra_params)
```
Comment (alorenzo175) [inline solarforecastarbiter/io/reference_observations/pvdaq.py:53]: missing from the combined function

File: solarforecastarbiter/io/reference_observations/tests/test_common.py [hunk 7]
```diff
@@ -776,3 +776,26 @@ def test_create_forecasts(template_fx, mocker):
...
+@pytest.mark.parametrize('params', [
+    {'network_api_id': 2},
+    {'network_api_id': '2'},
+])
+def test_apply_json_site_parameters_plant(test_json_site_file, params):
+    new_site = common.apply_json_site_parameters(
+        test_json_site_file,
+        {'extra_parameters': params},
+    )
+    assert 'modeling_parameters' in new_site
```
Comment (alorenzo175) [inline solarforecastarbiter/io/reference_observations/tests/test_common.py:790]: make sure extra_parameters are updated

**Pull request merged**